### PR TITLE
Boomerang (NOT READY TO PULL)

### PR DESCRIPTION
**Pull request by Camille92**

### CODE REVIEW

**Comment by ctubio** [comment]
:boom: :boom: :zap: :hospital:
this branch is still not ready for nowhere but in a dev box.

please cancel this PR as it will most likely break other existing quoting methods.

in case anyone want to have this branch merged into the great codes of michaelgrosner, please let me know in advance and i will try to avoid breaking other existing quoting methods.

btw double posting kills a kitty, everytime. your comment was answered in your first double post at https://github.com/ctubio/tribeca/commit/aa348bc7a795a73240bb714d971d01ac90072f07

**Comment by Camille92** [comment]
I have no idea how it came back here! But I try to save kitties as much as I can.
Tell me if you want so help debugging this branch and yes I think many people would be happy to have in the code as this strategy is more 'intelligent' than the others. (and going to mkt8 is really a plus).

On a side note I've seen you're from Bcn so if you want to have a caña to talk about it, feel free to tell me.

**Comment by michaelgrosner** [comment]
What is Boomerang?

**Comment by ctubio** [comment]
@Camille92 Thank you for the offer of the cañas but i don't drink beer 😞  maybe you smoke? that could be more convenient for setup a minimeeting with me 😄 

Boomerang is a new quoting mode, like PingPong, but works with all previous orders instead of with the latest.

For this, basic things has changed that may brake other quoting modes:
- Added 2 new columns to trades table (qty and price of matching order)
- Updated trades to be saved/displayed forever until a matching order is done (then can be cleaned).
- Load all records from database on startup without filtering by start_time.
- Removed almost all msgs to logs and also doesnt persist more than 1 record to some tables (attempt to allow a very very long run without crash in small hard disks).
- Added sounds and mods to frontend (diff sell/buy width input, clean button, new trade columns and removed some history panels or limited the amount of records displayed).
- Aaand.. maybe more things that i can't recall now (sorry but i didnt prepared/created this PR yet).

I'm sure all this can be conditional based on the quoting mode selected, so nothing breaks nothing, but currently i was in hurry for develop it and didnt care about other quoting modes. During this summer i can adapt it to make it work with other quoting modes (give me 2 month max).

I take no credit for this since i took the idea from the internets (see【BOT】C.A.T. Cryptocurrency Automatic Trader) and the sources from tribeca to make it work like it, therefore im happy to create again a new pull request anytime as soon as this is compatible with the rest of the features of the current official tribeca.

File: src/service/broker.ts [hunk 1]
```diff
@@ -168,6 +202,50 @@ export class OrderBroker implements Interfaces.IOrderBroker {
...
+          if (this._trades[i].tradeId==reTrade.tradeId) {
+            gowhile = true;
+            var allocQty = Math.min(trade.quantity, this._trades[i].quantity - this._trades[i].alloc);
+            this._trades[i].time = trade.time;
+            this._trades[i].allocprice = ((allocQty*trade.price) + (this._trades[i].alloc*this._trades[i].allocprice)) / (this._trades[i].alloc+allocQty);
+            this._trades[i].alloc += allocQty;
+            trade.quantity -= allocQty;
+            trade.value = Math.abs(trade.price*trade.quantity);
+            if (this._trades[i].quantity<=this._trades[i].alloc)
+              this._trades[i].value = Math.abs((this._trades[i].quantity*this._trades[i].price)-(this._trades[i].alloc*this._trades[i].allocprice));
```
Comment (ctubio) [inline src/service/broker.ts]: display the total income earned after a trade is closed

**Pull request closed without merging**